### PR TITLE
[add]implemented the user signup logic

### PR DESCRIPTION
**Pull request by Segun-Ade**

### CODE REVIEW

**Comment by justiceotuya** [comment]
Hi seun, Your code must travis build check before we can review. Please fix

**Comment by Segun-Ade** [comment]
@justiceotuya I am on it already.

**Comment by enyeInc** [comment]
@Segun-Ade your PR is failing the build step

**Comment by Segun-Ade** [comment]
@enyeInc yes, I see that it's failing to build because of this error 

> FirebaseError: projectId must be a string in FirebaseApp.options

>

File: src/components/signin/sagas.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as actions from './actions';
 // eslint-disable-next-line import/order
```
Comment (olajohn-ajiboye) [inline src/components/signin/sagas.js]: You should remove the eslint disable. Make sure you adhere to it.

File: src/components/signup/components/FormCard.jsx [hunk 2]
```diff
@@ -1,23 +1,93 @@
...
+const { GET_INPUT, REGISTER_WITH_EMAIL } = actionTypes;
+
+/**
+ *
+ * @param {*} key input value identified by a key
+ * @param {*} dispatch an action type GET_INPUT
+ * and pass the value from the key press event
+ * to payload
+ * @param {*} placeholder for each input field
+ * @param {*} inputType signifies the type for each input field
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: so when defining your params you have to say what type of object each param will be.
Exam
```
* @param {String} id - the user id
```

File: src/components/signup/components/FormCard.jsx [hunk 3]
```diff
@@ -1,23 +1,93 @@
...
+ */
+
+function InputField(key, dispatch, placeholder, inputType) {
+    return (
+        <Input
+            key={key}
+            onChange={e => dispatch({ payload: { key, value: e.target.value }, type: GET_INPUT })}
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: define you actions creators in a `actions.js` file.

File: src/components/signup/components/FormCard.jsx [hunk 4]
```diff
@@ -1,23 +1,93 @@
...
+            placeholder={placeholder}
+            type={inputType}
+        />
+    );
+}
+
+/**
+ *
+ * @param {*} dispatch an action type GET_INPUT
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: define param type

File: src/components/signup/components/FormCard.jsx [hunk 5]
```diff
@@ -1,23 +1,93 @@
...
-            <NavLink to="/">{SIGNIN}</NavLink>
-        </span>
-    </Form>
-);
+    );
+}
+
+/**
+ *
+ * @param {*} dispatch an action type REGISTER_WITH_EMAIL
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: same here

File: src/components/signup/components/FormCard.jsx [hunk 6]
```diff
@@ -1,23 +1,93 @@
...
+ * @param {*} info is a variable assigned to value
+ * of email, name and password in state
+ */
+
+function FormButton(dispatch, info) {
+    return (
+        <Form.Item>
+            <Button
+                className="button-p"
+                onClick={() => dispatch({ payload: info, type: REGISTER_WITH_EMAIL })}
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: define actions in `actions.js` file

File: src/components/signup/components/FormCard.jsx [hunk 7]
```diff
@@ -1,23 +1,93 @@
...
+ */
+export function FormCard() {
+    const dispatch = useDispatch();
+    const data = useSelector(state => state);
+    const { signup: { name, email, password } } = data;
+    const info = { email, name, password };
+    return (
+        <Form className="sign-form" layout="horizontal">
+            {FormItemInput(dispatch)}
+            {FormButton(dispatch, info)}
```
Comment (enyeInc) [inline src/components/signup/components/FormCard.jsx]: do not pass dispatch as parameter, define it directly in the components using it

File: src/components/signup/sagas.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    REGISTER_WITH_EMAIL,
+    REGISTRATION_ERROR,
+    REGISTRATION_SUCCESS,
+} = actionTypes;
+const { REACT_APP_USER_SIGNUP_API } = process.env;
+const api = REACT_APP_USER_SIGNUP_API;
+
+/**
+ *
+ * @param {*} payload is destructured from data
```
Comment (enyeInc) [inline src/components/signup/sagas.js]: deine param type

File: src/components/signup/sagas.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    const { email, password } = payload;
+    try {
+        const { user } = yield signUpWithEmail(email, password);
+        const userInfo = { userAuth: user, userType: 'user' };
+        const config = { headers: { 'Content-Type': 'application/json' } };
+        const body = JSON.stringify(userInfo);
+        const request = yield axios.post(api, body, config);
+        if (request.status === 200) {
+            const { status } = request.data;
+            yield put({ payload: status, type: REGISTRATION_SUCCESS });
```
Comment (enyeInc) [inline src/components/signup/sagas.js]: define action in `actions.js` file

File: src/firebase.js [hunk 10]
```diff
@@ -1,18 +1,29 @@
...
+export const firebaseApp = firebase.initializeApp(config);
+export const db = firebaseApp.firestore();
+
+export const auth = firebaseApp.auth();
+
+const provider = new firebase.auth.GoogleAuthProvider();
+provider.setCustomParameters({ prompt: 'select_account' });
+export const signUpWithGoogle = () => auth.signInWithPopup(provider);
+
+export const signUpWithEmail = async (email, password) => {
```
Comment (enyeInc) [inline src/firebase.js]: function needs comment

**Pull request merged**